File: TwoSum/TwoSum.js
var twoSum = function(nums, target) {
    let pairs = {};
    for(let n = 0; n < nums.length; n++)
    {
        let c = nums[n]
        if(c in pairs)
        {
            return [pairs[nums[n]], n];
        }
        let compliment = target - c;
        pairs[compliment] = n;
    }
    return -1;
};

/*
Equal to or fasther than 97.27% of Javascript submissions on LeetCode
https://leetcode.com/submissions/detail/144991438/
*/

File: TwoSum/TwoSum.py
class Solution:
    def twoSum(self, nums, target):
        pairs = {}
        for n in range(0, len(nums)):
            c = nums[n]
            if(c in pairs):
                return [pairs[nums[n]], n]
            compliment = target - c
            pairs[compliment] = n
        return -1

#
#Equal to or fasther than 100% of Python3 submissions on LeetCode
#https://leetcode.com/submissions/detail/154544795/
#